api/base/serializers.py

```diff
@@ -29,6 +29,65 @@ def _url_val(val, obj, serializer, **kwargs):
         return val
 
 
+class HyperlinkedIdentityFieldWithMeta(ser.HyperlinkedIdentityField):
+    # Returns a list with a url and an optional hash containing additional meta information
+
+    def __init__(self, view_name=None, **kwargs):
+        kwargs['read_only'] = True
+        kwargs['source'] = '*'
+        self.meta = kwargs.pop('meta', None)
+        super(ser.HyperlinkedIdentityField, self).__init__(view_name, **kwargs)
+
+    def get_url(self, obj, view_name, request, format):
+        """
+        Given an object, return the URL that hyperlinks to the object.
+
+        May raise a `NoReverseMatch` if the `view_name` and `lookup_field`
+        attributes are not configured to correctly match the URL conf.
+        """
+        # Unsaved objects will not yet have a valid URL.
+        if obj.pk is None:
+            return None
+
+        lookup_value = getattr(obj, self.lookup_field)
+
+        if lookup_value is None:
```

Comment:
This appears to be the only difference from the parent class's implementation. Was this added to avoid an error when `self.reverse` is called?